sdk/src/main/java/com/eegeo/mapapi/widgets/RouteViewHelper.java

```diff
@@ -0,0 +1,159 @@
+package com.eegeo.mapapi.widgets;
+
+import android.location.Location;
+import android.util.Pair;
+
+import com.eegeo.mapapi.geometry.LatLng;
+import com.eegeo.mapapi.services.routing.RouteStep;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+
+class RouteViewHelper {
+
+    public static boolean areApproximatelyEqual(LatLng firstLocation, LatLng secondLocation) {
+        final double epsilonSq = 1e-6;
+        float[] results = new float[1];
+        Location.distanceBetween(firstLocation.latitude, firstLocation.longitude, secondLocation.latitude, secondLocation.longitude, results);
+        return results[0] <= epsilonSq;
+    }
+
+    public static boolean areCoordinateElevationPairApproximatelyEqual(Pair<LatLng, Double> a, Pair<LatLng, Double> b) {
+        final double elevationEpsilon = 1e-3;
+
+        if(!areApproximatelyEqual(a.first, b.first)) {
+            return false;
+        }
+
+        return Math.abs(a.second - b.second) < elevationEpsilon;
+    }
+
+    public static void removeCoincidentPoints(List<LatLng> coordinates) {
+        if(coordinates == null) {
+            return;
+        }
+
+        Iterator<LatLng> it = coordinates.iterator();
+        LatLng prev = null;
+        LatLng current = it.next();
+        while (it.hasNext()) {
+            prev = current;
+            current = it.next();
+            if (areApproximatelyEqual(current, prev)) {
+                it.remove();
+            }
+        }
+    }
+
+    public static void removeCoincidentPointsWithElevations(List<LatLng> coordinates, List<Double> perPointElevations) {
+        List<Pair<LatLng, Double>> pairsList = new ArrayList<>(coordinates.size());
```

Comment:
Since we are removing consecutive points. Would it be better to use linked list here? As an array list would shift the entire list after removing any given point.

Reply:
implementation is changed for this method and in caller method we are using index based retrieve.